providers/oauthProvider.js

```diff
@@ -130,3 +132,35 @@ exports.makeUserSiteAdmin = (externalUserId, oauthDefaultId) => {
       body: JSON.stringify(body)
     });
 };
+
+
+exports.copyUsersFromSite = async (oldSiteId, newSiteId) => {
+  const roleResponse = await userRoleApi.fetchAll();
+  const roleMap = new Map(roleResponse.map(role => [role.name, role.id]));
+  const newSite = await siteApi.fetch(newSiteId);
+  const newClientId = newSite.config?.oauth?.default?.id;
+  const usersOfOldSite = await siteApi.fetchUsersOfSite(oldSiteId);
+
+  const requests = usersOfOldSite.map((user) => {
+  const mappedRoleId = roleMap.get(user.role);
+  console.log(`Adding userrole to client with id: ${newClientId}, externalUserId: ${user.externalUserId} and roleId: ${mappedRoleId}`)
+
+  const url = process.env.USER_API + '/api/admin/user/' + user.externalUserId;
+  const body = {
+    client_id: process.env.USER_API_CLIENT_ID,
+    client_secret: process.env.USER_API_CLIENT_SECRET,
+    roles: {},
+  };
+
+  body.roles[newClientId] = mappedRoleId;
+  return fetch(
```

Comment:
I don't have a problem with direct api calls, but it would be more consistent to use 
the userService as with all other requests done here.